src/tape/ensemble.py

```diff
@@ -1847,6 +1847,36 @@ def _sync_tables(self):
         self.object.set_dirty(False)
         return self
 
+    def select_random_timeseries(self, seed=None):
+        """Selects a random lightcurve from the Ensemble
+
+        Parameters
+        ----------
+        seed: int, or None
+            Sets a seed to return the same object id on successive runs. `None`
+            by default, in which case a seed is not set for the operation.
+
+        Returns
+        -------
+        ts: `TimeSeries`
+            Timeseries for a single object
+
+        """
+
+        if seed is not None:
+            np.random.seed(seed)
+
+        # Avoid a choice from full index space, select a random partition to grab from
+        if self.object.npartitions > 1:
+            partition_num = np.random.randint(0, self.object.npartitions - 1)
```

Comment:
We never select the last partition here, because `np.random.randint`'s second argument is an exclusive high value. Actually, I'm not sure we even need `if-else` here: for `self.object.npartitions == 1` `randint(0, 1)` will always return zero.

Reply:
Should be addressed by the refactor done in the latest commit